### PR TITLE
Add labels to Curriculum Catalog Card

### PR DESCRIPTION
**Pull request by megcrenshaw**
This PR does two things:
1) Adds new tag styling as indicated in the [Redline](https://www.figma.com/file/w5R6KSZF01FAysZor6j4sa/Curriculum-Catalog?node-id=101-9&t=xDEHAF1BxmOB2Qkw-0). Instructions on what to do on overflow are in [this thread](https://codedotorg.slack.com/archives/C0SUN2SSF/p1681388786515459?thread_ts=1681332665.798969&cid=C0SUN2SSF). I did need to make the cards longer to accommodate the extra space needed for the tags.
2) Ensures tags are in the ordered specified in the [product spec](https://docs.google.com/document/d/1UID-xzCF5gxOHihcYiYuZwpotCpGXGPTzqX1UyS37-I/edit#bookmark=id.79qjm3vrwpqg).

From Redline:
image

On webpage:
image

Two-line title:
image

No topics or subjects (no picture loaded yet):
image

Ellipse:
image

Non-English:
image

With hover (but will update to use Tooltip):
https://user-images.githubusercontent.com/9142121/232040244-051289df-e7af-40bd-972a-aa5c5fe59e15.mov

## Follow-up work

Tooltip on hover for both the first tag and the remaining tags: https://codedotorg.atlassian.net/browse/ACQ-524

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by megcrenshaw** [comment]
Need to change the hover styling ... will re-request once that's implemented

**Comment by megcrenshaw** [comment]
Implementing the tooltip is going to require more work to make it accessible. I'm splitting that out into a separate ticket because this PR is big enough. https://codedotorg.atlassian.net/browse/ACQ-524

**Comment by kelbyhawn** [comment]
This looks great! Will the tooltip list out all of the topics stacked vertically?

**Comment by davidsbailey** [comment]
> This PR does two things:
> 
> 1. Adds new tag styling as indicated in the [Redline](https://www.figma.com/file/w5R6KSZF01FAysZor6j4sa/Curriculum-Catalog?node-id=101-9&t=xDEHAF1BxmOB2Qkw-0). Instructions on what to do on overflow are in [this thread](https://codedotorg.slack.com/archives/C0SUN2SSF/p1681388786515459?thread_ts=1681332665.798969&cid=C0SUN2SSF). I did need to make the cards longer to accommodate the extra space needed for the tags.
> 2. Ensures tags are in the ordered specified in the [product spec](https://docs.google.com/document/d/1UID-xzCF5gxOHihcYiYuZwpotCpGXGPTzqX1UyS37-I/edit#bookmark=id.79qjm3vrwpqg).

sorry but what are tags in this context?

**Comment by megcrenshaw** [comment]
> This looks great! Will the tooltip list out all of the topics stacked vertically?

Yes! The first pass of that looks like (just updated the follow-up work section to make this clear)
image (2)

**Comment by megcrenshaw** [comment]
> > This PR does two things:
> > 
> > 1. Adds new tag styling as indicated in the [Redline](https://www.figma.com/file/w5R6KSZF01FAysZor6j4sa/Curriculum-Catalog?node-id=101-9&t=xDEHAF1BxmOB2Qkw-0). Instructions on what to do on overflow are in [this thread](https://codedotorg.slack.com/archives/C0SUN2SSF/p1681388786515459?thread_ts=1681332665.798969&cid=C0SUN2SSF). I did need to make the cards longer to accommodate the extra space needed for the tags.
> > 2. Ensures tags are in the ordered specified in the [product spec](https://docs.google.com/document/d/1UID-xzCF5gxOHihcYiYuZwpotCpGXGPTzqX1UyS37-I/edit#bookmark=id.79qjm3vrwpqg).
> 
> sorry but what are tags in this context?

Oh yikes! I've just been calling these `tags` in my head for forever for no real reason. Tags are the blue "bubbles" underneath the image that show a category name –– they could be a `subject` or a `topic` from the backend. I'm only using the term `tag` to describe the styles, but would `category` be a more clear name here? I hesitate to do that because `category` is a separate field on the backend.

Another potential follow-up is that on the backend, these fields are called `cs_topic` and `school_subject` which is misleading because there could be multiple topics and subjects, formatted as a CSV. Adding some validation here would be nice. Do you think it also warrants a name change to the fields to be clear that there could be multiple?

**Comment by davidsbailey** [comment]
> > sorry but what are tags in this context?
> 
> Oh yikes! I've just been calling these `tags` in my head for forever for no real reason. Tags are the blue "bubbles" underneath the image that show a category name –– they could be a `subject` or a `topic` from the backend. I'm only using the term `tag` to describe the styles, but would `category` be a more clear name here? I hesitate to do that because `category` is a separate field on the backend.

No worries! in this comment, I was mostly asking to help figure out what I was reviewing before digging in.

having read through a bit, I would suggest either "label", which sounds self-documenting to me, or keeping "tag" and adding a well-placed comment indicating what it means.

> Another potential follow-up is that on the backend, these fields are called `cs_topic` and `school_subject` which is misleading because there could be multiple topics and subjects, formatted as a CSV. Adding some validation here would be nice. Do you think it also warrants a name change to the fields to be clear that there could be multiple?

if I'm understanding correctly, the change would be to rename these fields to `cs_topics` and `school_subjects` on the backend and in the data that gets sent to the client? yes, I think that would be a good idea, while this is still unlaunched and therefore easier to change.

**Review by davidsbailey — APPROVED** [review]
LGTM after adding a brief comment on what a tag is, or please request re-review if doing a bigger change like switching to "label".

**Comment by megcrenshaw** [comment]
> LGTM after adding a brief comment on what a tag is, or please request re-review if doing a bigger change like switching to "label".

I like the `labels` change!
- change from tag to labels: https://github.com/code-dot-org/code-dot-org/pull/51304/commits/5439436a32e919776d086e27acb8e1225b86a598
- Other commits can be seen in the history, but I did rebase due to merge conflicts. I also changed styling to use child selectors rather than classnames (https://github.com/code-dot-org/code-dot-org/pull/51304/commits/47ed60d5669f125b7c38ad2dc644bf8bd2b6d759), and there was a bug when no labels were present, so I fixed that (https://github.com/code-dot-org/code-dot-org/pull/51304/commits/cf67a043c10fb47118b28a53a84215a5462a17f0) and added another story (https://github.com/code-dot-org/code-dot-org/pull/51304/commits/cb498cdf638898cde34e1274c2a7294b71389687)

File: apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss
Comment (kelbyhawn) [inline apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss:1]: Not blocking, but I just noticed the non-English stacked buttons height (on line 127) `.buttonsContainer_notEnglish` doesn't match the English button height — both should be `height: 40px`.

**Review by kelbyhawn — APPROVED** [review]
Looks great, and I like the tags >> labels change!

File: apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx [hunk 1]
```diff
@@ -39,10 +55,58 @@ describe('CurriculumCatalogCard', () => {
...
+        topics={topics}
+      />
+    );
+
+    screen.getByText(subjectsAndTopicsOrder[firstSubjectIndexUsed], {
+      exact: false
+    });
+  });
+
+  it('renders tag of number of remaining subjects and topics when there is more than one present', () => {
```
Comment (davidsbailey) [inline apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx]: tag -> label

File: apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx [hunk 2]
```diff
@@ -39,10 +55,58 @@ describe('CurriculumCatalogCard', () => {
...
+
+  it('renders one topic, the first available from ordered list, if no subject present', () => {
+    render(<CurriculumCatalogCard {...defaultProps} topics={topics} />);
+
+    screen.getByText(subjectsAndTopicsOrder[firstTopicIndexUsed], {
+      exact: false
+    });
+  });
+
+  it('does not render number of remaining tag when exactly one subject or one topic is present', () => {
```
Comment (davidsbailey) [inline apps/test/unit/templates/curriculumCatalog/curriculumCatalogCardTest.jsx]: tag -> label

**Review by davidsbailey — APPROVED** [review]
looks great, thanks for updating! just two tiny nits below

File: apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss
Comment (megcrenshaw) [reply apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss:1]: This was actually intentional ... if it's larger, the card needs to be even longer to have some space between the grades and the button
image

Here's what it looks like now
image

Comment (megcrenshaw) [reply apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss:1]: I'll ask in the design channel about this!

Comment (megcrenshaw) [reply apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss:1]: Actually I just made the executive decision to change it with a larger height

image

Comment (kelbyhawn) [reply apps/src/templates/curriculumCatalog/curriculum_catalog_card.module.scss:1]: Nice! I think this was the right call 👍

**Pull request merged**